src/components/IFrame.js

```diff
@@ -27,76 +37,236 @@ function getNewDotURL(url) {
+    }
+
+    if (screenName === SCREENS.REPORTS_OLDDOT) {
+        return 'reports';
+    }
+
+    return undefined;
 }
 
-export default function ReportScreen() {
-    const [oldDotURL, setOldDotURL] = useState('https://www.expensify.com.dev');
+const propTypes = {
+    // The session of the logged in person
+    session: PropTypes.shape({
+        // The email of the logged in person
+        email: PropTypes.string,
 
-    useEffect(() => {
-        setOldDotURL(`https://expensify.com.dev/${getOldDotURL(window.location.href)}`);
+        // The authToken of the logged in person
+        authToken: PropTypes.string,
+    }).isRequired,
+};
+
+// Gets the routeName and params of the top most central pane route if it is an iframe route.
+// If the top most central pane route is not an iframe route, then it returns undefined.
+function getIframeRouteNameAndParams(state) {
+    const routeName = getTopMostCentralPaneRouteName(state);
+    if (routeName !== 'Report') {
+        const topMostCentralPane = lodashFindLast(state.routes, (route) => route.name === 'CentralPaneNavigator');
+        if (topMostCentralPane.state) {
+            const params = topMostCentralPane.state.routes[topMostCentralPane.state.index || 0].params;
+            return {routeName, params};
+        }
+
+        const params = topMostCentralPane.params;
+        if (params) {
+            return {routeName, params: params.params};
+        }
+    }
+    return {routeName: undefined, params: undefined};
+}
+
+// TODO: use proper URL
+const BASE_IFRAME_URL = 'http://localhost:3000';
+
+function IFrame({session}) {
+    const [oldDotURL, setOldDotURL] = useState(undefined);
+    const {routeName, params} = useNavigationState((state) => getIframeRouteNameAndParams(state));
 
+    useEffect(() => {
         window.addEventListener('message', (event) => {
-            const url = event.data;
+            let data = {};
+            try {
+                data = JSON.parse(event.data);
+            } catch {
+                data = {};
+            }
+
+            if (data.iframeId !== 'OldDot' || data.type !== 'URL_CHANGED') {
+                return;
+            }
+
+            const {url} = data.payload ?? {};
+
             // TODO: use this value to navigate to a new path
             // eslint-disable-next-line no-unused-vars
             const newDotURL = getNewDotURL(url);
         });
     }, []);
 
+    useEffect(() => {
+        // The routeName will be undefined if there is no iframe route on the top.
+        // In that case we don't want to change the iframe URL.
+        if (!routeName) {
+            return;
+        }
+
+        const newOldDotURL = getOldDotURL(routeName, params);
+        if (!newOldDotURL) {
+            console.error(`No old dot URL found for routeName: ${routeName} and params: ${JSON.stringify(params)}`);
+            return;
+        }
+
+        setOldDotURL(`${BASE_IFRAME_URL}/${newOldDotURL}`);
```

Comment:
I did some testing and it turns out that we need to remount the iframe anyway. Keeping the iframe and navigating inside it will interfere with the browser history. 

I did it in a way to remount as little as possible.

Maybe there is a way to prevent iframe from interfering with the browser history without remounting but I can't see anything at this moment. That's probably something worth investigating in the future